client/src/Pages/SignUp.tsx

```diff
@@ -37,7 +40,7 @@ function SignUp() {
     },
   ];
 
-  return (
+  return (userInfo?.user.providerID ? (<>{navigate(`/user/${userInfo?.user.providerID}`)}</>) : (
```

Comment:
I think it's better to do this in AuthLAyout to redirect directly, instead of doing this twice
```jsx
 return userInfo?.user?.providerID ? <Navigate to={`/user/${userInfo?.user.providerID}`} replace /> : (
    <>
      <Nav language={lang} setLanguage={setLang} />
      <Outlet />
    </>
```